### DISCUSSION TITLE
from_model_instance and adding extra fields

### DISCUSSION THREAD

**Discussion by ste-pool** [litestar-org/litestar#1545] [Q&A]
EDIT: Note that I'm still running under starlite and haven't migrated to litestar yet...

There's probably something obvious I'm missing or not understanding...

In the documentation, there is an example:

https://docs.litestar.dev/latest/usage/dto.html#add-new-fields

```python
from pydantic import BaseModel

from starlite import DTOFactory


class MyClass(BaseModel):
    first: int
    second: int


dto_factory = DTOFactory()

MyClassDTO = dto_factory("MyClassDTO", MyClass, field_definitions={"third": (str, ...)})
```
However, you can never use the from_model_instance on the DTO as it has a required field that can't be injected:
```python
x = MyClass(first=1, second=2)
y = MyClassDTO.from_model_instance(x)
...
>>> pydantic.error_wrappers.ValidationError: 1 validation error for MyClassDTO
>>> third                                                                     
>>>  field required (type=value_error.missing)                               
```

Is there a way to add extra arguments without making them optional? Something like:
```
y = MyClassDTO.from_model_instance(x, third=3)
```

For context, my model is an sqlalchemy table and I would like to add some calculated fields before being returned to the route. They don't have default values and I'd like to make them required so that they are documented/validated correctly.

**Comment by JacobCoffee** [comment]
@peterschutt do you have an idea here? Trying to go through older issues.
